hw/ip/otbn/doc/_index.md

```diff
@@ -361,6 +361,66 @@ Both use the same state for tracking control flow.
 This is a stack of tuples containing a loop count, start address and end address.
 The stack has a maximum depth of eight and the top of the stack is the current loop.
 
+### Loop nesting
+
+OTBN permits nested loops and branch and jumps inside loops.
+The programmer must ensure loops are *well nested*.
```

Comment:
This is too waffly for me. The programmer can fail to do so, and everything might still work just fine. Can we phrase it as "In order for X to hold, the programmer must ensure that loops are well nested."?

Alternatively (and, I think, better), the text could be rephrased to be something more like this:
```
OTBN permits loop nesting and branches and jumps inside loops.
However, it doesn't have support for early termination of loops: there's no way to pop an entry from the loop stack without executing the last instruction of the loop the correct number of times.
It can also only pop one level of the loop stack per instruction.

To avoid polluting the loop stack, the programmer must ensure that:
- Even if there are branches and jumps within a loop body, the final instruction of the loop body gets executed exactly once per iteration.
- Nested loops have distinct end addresses.
```
and then give the (excellent) examples.